### PR TITLE
[ATO-706] Add documentation for secrets managers

### PR DESCRIPTION
**Pull request by radovanZRasa**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by m-vdb** [comment]
@radovanZRasa I've resquested reviews from Bart and Lena instead of Arjaan :)

File: docs/sidebars.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ module.exports = {
                     label: 'Architecture', // name still confusing with architecture page elsewhere
                     items: [
                         'arch-overview',
+                        'secrets-managers',
```
Comment (m-vdb) [inline docs/sidebars.js:175]: I would put it below `lock-stores`

File: docs/sidebars.js [hunk 2]
```diff
@@ -172,6 +172,7 @@ module.exports = {
                     label: 'Architecture', // name still confusing with architecture page elsewhere
                     items: [
                         'arch-overview',
```
Comment (m-vdb) [inline docs/sidebars.js:174]: the architecture overview will probably become outdated BTW. It's related to [this ticket](https://rasahq.atlassian.net/browse/ATO-483) but not quite the same

File: docs/docs/secrets-managers.mdx [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+id: secrets-managers
+sidebar_label: Secrets Managers
+title: Secrets Managers
+description: Credentials used to authenticate to external services are safely stored in a secrets manager.
+abstract: Credentials managers are used to safeguard credentials used to authenticate to external services.
```
[the remark this answers is not in the document]
Comment (m-vdb) [reply docs/docs/secrets-managers.mdx:6]: I would rephrase to:

> You can store your assistant's secrets in an external credentials manager. Rasa Pro currently supports credentials manager for the Tracker Store (link to the tracker store doc page).

File: docs/docs/secrets-managers.mdx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+## Secrets Managers
+Secrets managers are used to safeguard credentials used to authenticate to external services,
+like databases.
+
+Rasa Pro supports the following secrets managers:
+* [HashiCorp Vault](https://www.hashicorp.com/products/vault)
+
+
+Currently, Rasa Pro supports sefeguarding credentials for the following services:
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:24]:
```suggestion
Currently, Rasa Pro supports safeguarding credentials for the following services:
```

File: docs/docs/secrets-managers.mdx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+- [Tracker Stores](./tracker-stores.mdx)
+
+
+## Vault Secrets Manager
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:28]:
```suggestion
## HashiCorp Vault Secrets Manager
```

Comment (m-vdb) [reply docs/docs/secrets-managers.mdx:28]: I think we need consistent naming for this one too

File: docs/docs/secrets-managers.mdx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+The Vault Secrets Manager is used to store credentials used to authenticate to external services.
+The credentials are stored in a Vault instance and can be encrypted at rest.
+
+If you need to encrypt your credentials,
+you can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
+
+:::note
+Expiring tokens need to be renewed periodically, and renewal process is done over the network,
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:37]:
```suggestion
Expiring tokens need to be renewed periodically, and the renewal process is done over the network,
```

File: docs/docs/secrets-managers.mdx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+we will try to renew the token 5 seconds before it expires.
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:38]:
```suggestion
Rasa Pro will try to renew the token 5 seconds before it expires.
```

File: docs/docs/secrets-managers.mdx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+If token's TTL is less than 5 seconds, we will try to renew it ASAP,
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:39]:
```suggestion
If the token's time-to-live (TTL) is less than 5 seconds, we will try to renew it as soon as possible,
```

Comment (m-vdb) [reply docs/docs/secrets-managers.mdx:39]: what does "as soon as possible" mean in this context?

File: docs/docs/secrets-managers.mdx [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+but it might fail due to network latency.
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:40]: I think we need a suggestion / recommendation here. What happens if it fails, what will users need to do? Do we recommend TTL > 5s ? What does HashiCorp recommend?

File: docs/docs/secrets-managers.mdx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+:::
+
+
+### Authentication
+Rasa Pro supports authentication to Vault through
+[Token](https://www.vaultproject.io/docs/auth/token).
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:46]:
```suggestion
Rasa Pro supports connections to Vault via
[Token authentication](https://www.vaultproject.io/docs/auth/token).
```

File: docs/docs/secrets-managers.mdx [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+Rasa Pro will automatically renew the token if it is expiring.
+
+
+### How to configure access to Vault
+
+Access to Vault secrets manager can be configured with environment variables
+and through `endpoints.yml` configuration file.
+
+Environment variables and `endpoints.yml` configuration file are merged together
+and *the values from the environment variables take precedence*.
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:58]: nice -- would be great to know if this is a pattern that we strictly follow in other Rasa features cc @madebymagnolia

File: docs/docs/secrets-managers.mdx [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+Following environment variables are available:
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:60]:
```suggestion
The following environment variables are available:
```

File: docs/docs/secrets-managers.mdx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+```bash
+SECRET_MANAGER = "vault" # which secrets manager to use, currently only vault is supported
+VAULT_HOST # address of the vault server
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:66]:
```suggestion
SECRET_MANAGER="vault" # which secrets manager to use, currently only 'vault' is supported
VAULT_HOST="<host>" # address of the vault server
VAULT_TOKEN="<token>" # token to authenticate to the vault server
VAULT_RASA_SECRETS_PATH="<path>" # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
VAULT_TRANSIT_MOUNT_POINT="<mount-point>" # if transit secrets engine is enabled, set this to mount point of the transit engine
```

File: docs/docs/secrets-managers.mdx [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
+    type: vault
+    token: <token used to authenticate to Vault>
+    url: "http://localhost:1234"
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:74]: `url` here and `VAULT_HOST` above are kind of confusing. Can't we unify the variable names? (pick url or host)

File: docs/docs/secrets-managers.mdx [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+SECRET_MANAGER = "vault" # which secrets manager to use, currently only vault is supported
+VAULT_HOST # address of the vault server
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:71]: I would add exactly the same comments. If users look at the doc, they might just look at the yml file doc if this is their preferred way, and will have to scroll up to look at the doc for each variable. (I'm talking about comments like "# which secrets manager to use, currently only 'vault' is supported")

File: docs/docs/secrets-managers.mdx [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+    type: vault
+    token: <token used to authenticate to Vault>
+    url: "http://localhost:1234"
+    secrets_path: rasa-secrets  # if not set it defaults to `rasa-secrets`
+    transit_mount_point: transit # if you have enabled transit secrets engine, and you want to use it
+```
+
+#### How to combine environment variables and `endpoints.yml`
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:79]: I think the wording here needs to be clearer and say that we recommend storing sensitive information in the environment variables instead of files. WDYT?

File: docs/docs/secrets-managers.mdx [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+A simple example on how to combine environment variables and `endpoints.yml` configuration file
+would be to store access token in the environment variable and the rest of the configuration
+in the `endpoints.yml` file.
+
+```bash
+# environment variables
+VAULT_TOKEN=<token used to authenticate to Vault>
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:86]:
```suggestion
VAULT_TOKEN="<token>" # token to authenticate to the vault server
```
(consistent with what's written above)

File: docs/docs/secrets-managers.mdx [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+
+```yaml-rasa {3} title="endpoints.yml
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:90]: is the `{3}` highlight intended? I don't think so 🤔

File: docs/docs/secrets-managers.mdx [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+```bash
+SECRET_MANAGER = "vault" # which secrets manager to use, currently only vault is supported
+VAULT_HOST # address of the vault server
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
+```yaml-rasa {3} title="endpoints.yml
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:70]: is the `{3}` highlight intended? I don't think so 🤔

File: docs/docs/secrets-managers.mdx [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+secrets_manager:
+    type: vault
+    url: "http://localhost:1234"
+    secrets_path: rasa-secrets  # if not set it defaults to `rasa-secrets`
+    transit_mount_point: transit # if you have enabled transit secrets engine, and you want to use it
+```
+
+### Example
+This is a simple example which demonstrates how to store credentials for tracker store in Vault
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:99]:
```suggestion
This is a simple example which demonstrates how to store credentials for the Tracker Store in Vault
```

File: docs/docs/secrets-managers.mdx [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+    ```bash
+    docker run --name postgres -e POSTGRES_DB=rasa -e POSTGRES_PASSWORD=12345 -p 5432:5432 -d postgres
+    ```
+
+4. Configure Rasa to access the Vault instance
+
+    Checkout the [How to configure access to Vault](#how-to-configure-access-to-vault) section for more details.
+
+5. Configure Rasa to use the Vault secrets manager to fetch credentials for the tracker store
+    ```yaml-rasa {3} title="endpoints.yml
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:130]: is the `{3}` highlight intended? I don't think so 🤔

File: docs/docs/secrets-managers.mdx [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
+    type: vault
+    url: "http://localhost:1234"
+    secrets_path: rasa-secrets  # if not set it defaults to `rasa-secrets`
+    transit_mount_point: transit # if you have enabled transit secrets engine, and you want to use it
+```
+
+### Example
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:98]: I would rework this section and add 2 sections:
- one about Configuring the tracker store and showing how to write the yml file  (line 130)
- one about an Example -> I would check with CSE if the example you wrote down is good enough for our customers

**Review by m-vdb — CHANGES_REQUESTED** [review]

File: docs/docs/secrets-managers.mdx [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import RasaProLabel from '@theme/RasaProLabel';
+import RasaProBanner from '@theme/RasaProBanner';
+
+<RasaProLabel />
+<RasaProBanner/>
+
+## Secrets Managers
+Secrets managers are used to safeguard credentials used to authenticate to external services,
```
Comment (ancalita) [inline docs/docs/secrets-managers.mdx:17]: As a general rule of thumbs, avoid passive voice as much as possible:
```suggestion
Secrets managers safeguard credentials used to authenticate access to external services,
```

File: docs/docs/secrets-managers.mdx [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+* [HashiCorp Vault](https://www.hashicorp.com/products/vault)
+
+
+Currently, Rasa Pro supports sefeguarding credentials for the following services:
+- [Tracker Stores](./tracker-stores.mdx)
+
+
+## Vault Secrets Manager
+
+The Vault Secrets Manager is used to store credentials used to authenticate to external services.
```
Comment (ancalita) [inline docs/docs/secrets-managers.mdx:30]:
```suggestion
The Vault Secrets Manager is used to store credentials used to authenticate access to external services.
```

File: docs/docs/secrets-managers.mdx [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+The credentials are stored in a Vault instance and can be encrypted at rest.
```
Comment (ancalita) [inline docs/docs/secrets-managers.mdx:31]: We should rephrase to active voice and provide a link to Vault documentation/tutorial on how to store credentials in a Vault instance, e.g. `To store credentials in a Vault instance, you can read the official Vault docs. You can also encrypt credentials at rest.`

File: docs/docs/secrets-managers.mdx [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+Following environment variables are available:
+```bash
+SECRET_MANAGER = "vault" # which secrets manager to use, currently only vault is supported
+VAULT_HOST # address of the vault server
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
```
Comment (ancalita) [inline docs/docs/secrets-managers.mdx:69]:
```suggestion
To configure the Vault secrets manager, you can fill the following section in `endpoints.yml` file:
```

File: docs/docs/secrets-managers.mdx [hunk 27]
```diff
@@ -0,0 +1,142 @@
...
+### How to configure access to Vault
+
+Access to Vault secrets manager can be configured with environment variables
+and through `endpoints.yml` configuration file.
+
+Environment variables and `endpoints.yml` configuration file are merged together
+and *the values from the environment variables take precedence*.
+
+Following environment variables are available:
+```bash
```
Comment (l2nguyen) [inline docs/docs/secrets-managers.mdx:61]: | Environment Variable          | Description                                                                                    | Default               |
| :---------------------------- | :----------------------------------------------------------------------------------------------| :-------------------- |
| `SECRET_MANAGER`              | **Required**. The secrets manager to use. *Currently only "vault" is supported*                | `vault`               |
| `VAULT_HOST`                  | **Required**. The address of the vault server                                                  |                       |
| `VAULT_TOKEN`                 | **Required**. token to authenticate to the vault server                                        |                       |
| `VAULT_RASA_SECRETS_PATH`     | Path to the secrets in the vault server                                                        | `rasa-secrets`        |
| `VAULT_TRANSIT_MOUNT_POINT`   | If transit secrets engine is enabled, set this to  mount point of the transit engine           |                       |

File: docs/docs/secrets-managers.mdx [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+              url: localhost:5432
+              username:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_username
+              password:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_password
+    ```
+
+6. Start Rasa
```
Comment (Balowen) [inline docs/docs/secrets-managers.mdx:142]: What do we mean here by `Start Rasa`? Does it mean Rasa server needs to be shut down when adding secrets to the HashiCorp Vault? Will the changes in endpoint.yml get applied if the user restarts rasa pod? I'm thinking here about our customers that would want to implement HashiCorp Vault into their current deployment

File: docs/docs/secrets-managers.mdx (hunk 8 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:39]: Now I have realized that we have not covered this use case in the best way. 
If token's TTL is less than 5 seconds we should renew the token right away instead of waiting for token's remaining TTL.

File: docs/docs/secrets-managers.mdx (hunk 28 above)
Comment (Balowen) [reply docs/docs/secrets-managers.mdx:142]: We should be more specific here, this example is essential to set up the Vault and use it properly.

File: docs/docs/secrets-managers.mdx (hunk 8 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:39]: I opened [PR](https://github.com/RasaHQ/rasa-plus/pull/365) to address this situation.

Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:39]: After aforementioned PR is merged, we can say that it will be renewed after 1 second.

File: docs/docs/secrets-managers.mdx (hunk 9 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:40]: > What happens if it fails, what will users need to do?
Users will have to create a new token and restart Rasa with that new token.

We can build re-try mechanism for renewal of tokens.

HashiCorp does not have general recommendation for Vault token's TTL.
We can say that we recommend that token's TTL to be at least 1 minute.

File: docs/docs/secrets-managers.mdx (hunk 10 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:46]: I would recommend we use:

```
Rasa Pro can authenticate to Vault through
[Token authentication](https://www.vaultproject.io/docs/auth/token).
````

because Rasa will authenticate to Vault through token. To say that it connects to Vault through token authentication sounds a bit mixed to me.

File: docs/sidebars.js (hunk 2 above)
Comment (m-vdb) [reply docs/sidebars.js:174]: yes, defintely, but we need a ticket in our backlog. Didn't have the time to create it yet

File: docs/docs/secrets-managers.mdx (hunk 14 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:74]: Opened [PR](https://github.com/RasaHQ/rasa-plus/pull/366) to address change in the code

File: docs/docs/secrets-managers.mdx (hunk 16 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:79]: How does:
```
####  Store access credentials in environment variables
```
sound?

File: docs/docs/secrets-managers.mdx (hunk 22 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:98]: @Balowen @l2nguyen What do you think about the example written here?
Should we keep it/improve it/remove it?

Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:98]: > I would rework this section and add 2 sections:
> 
>     * one about Configuring the tracker store and showing how to write the yml file  (line 130)

I can add:

### How to configure Tracker Store to use Vault Secrets Manager
1. Configure Rasa to access the Vault instance

    Checkout the [How to configure access to Vault](#how-to-configure-access-to-vault) section for more details.

2. Configure Rasa to use the Vault secrets manager to fetch credentials for the tracker store
    ```yaml-rasa title="endpoints.yml
            tracker_store:
                  type: SQL
                  url: localhost:5432
                  username:
                        source: secrets_manager.vault
                        secret_key: sql_store_username
                  password:
                        source: secrets_manager.vault
                        secret_key: sql_store_password
    ```

WDYT?

File: docs/docs/secrets-managers.mdx (hunk 28 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:142]: Rasa does not need to be stopped when secrets to Vault are added.
Rasa only needs to be stopped when we are adding/updating Vault connection configuration as Rasa currently does not support hot-reloading of it's configuration.

File: docs/docs/secrets-managers.mdx (hunk 27 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:61]: To keep env vars and endpoints.yml consistent WDYT about:

Toe following environment variables are available:
| Environment Variable          | Description                                                                                    | Default               |
| :---------------------------- | :----------------------------------------------------------------------------------------------| :-------------------- |
| `SECRET_MANAGER`              | **Required**. The secrets manager to use. *Currently only "vault" is supported*                | `vault`               |
| `VAULT_HOST`                  | **Required**. The address of the vault server                                                  |                       |
| `VAULT_TOKEN`                 | **Required**. token to authenticate to the vault server                                        |                       |
| `VAULT_RASA_SECRETS_PATH`     | Path to the secrets in the vault server                                                        | `rasa-secrets`        |
| `VAULT_TRANSIT_MOUNT_POINT`   | If transit secrets engine is enabled, set this to mount point of the transit engine            |                       |

To configure the Vault secrets manager, you can fill the following section in `endpoints.yml` file:
```yaml-rasa title="endpoints.yml
secrets_manager:
    type: vault # required - the secrets manager to use
    token: <token used to authenticate to Vault> # required - token to authenticate to the vault server   
    url: "http://localhost:1234" # required - the address of the vault server      
    secrets_path: rasa-secrets  # path to the secrets in the vault server if not set it defaults to `rasa-secrets`
    transit_mount_point: transit # if transit secrets engine is enabled, set this to mount point of the transit engine
```

File: docs/sidebars.js (hunk 2 above)
Comment (radovanZRasa) [reply docs/sidebars.js]: Added [Adapt Architecture Overview page with user tailored content](https://rasahq.atlassian.net/browse/ATO-834)

File: docs/docs/secrets-managers.mdx [hunk 29]
```diff
@@ -0,0 +1,170 @@
...
+```
+
+### How to configure Tracker Store to read authentication credentials from Vault Secrets Manager
+1. Configure Rasa to access the Vault instance
+
+    Checkout the [How to configure access to Vault](#how-to-configure-access-to-vault) section for more details.
+
+2. Configure Rasa to use the Vault secrets manager to fetch credentials for the tracker store
+    ```yaml-rasa title="endpoints.yml
+            tracker_store:
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:114]: top level indentation probably needs to be reset

File: docs/docs/secrets-managers.mdx [hunk 30]
```diff
@@ -0,0 +1,170 @@
...
+        tracker_store:
+              type: SQL
+              url: localhost:5432
+              username:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_username
+              password:
+                    source: secrets_manager.vault
+                    secret_key: sql_store_password
+    ```
```
Comment (m-vdb) [inline docs/docs/secrets-managers.mdx:168]: this duplicates the content above, I would remove and link to it

File: docs/docs/secrets-managers.mdx (hunk 9 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:43]: I opened [PR](https://github.com/RasaHQ/rasa-plus/pull/372) to address this.

To accommodate for network latency I increased threshold for renewal so that it starts 15 seconds before token will expire and retry 5 times every second if renewal fails.

File: docs/docs/secrets-managers.mdx (hunk 13 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx]: Addressed in [comment](https://github.com/RasaHQ/rasa/pull/12110#discussion_r1130770521)

File: docs/docs/secrets-managers.mdx [hunk 31]
```diff
@@ -0,0 +1,170 @@
...
+---
+
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import RasaProLabel from '@theme/RasaProLabel';
+import RasaProBanner from '@theme/RasaProBanner';
+
+<RasaProLabel />
+<RasaProBanner/>
+
+## Secrets Managers
```
[the remark this answers is not in the document]
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:17]: I will remove the title and sentence below it.
We should list which secret managers we support and in which service (with links for easier navigation).

File: docs/docs/secrets-managers.mdx [hunk 32]
```diff
@@ -0,0 +1,152 @@
...
+
+## HashiCorp Vault Secrets Manager
+
+Use Vault Secrets Manager to store credentials used to authenticate access to external services.
+The credentials are stored in a Vault instance and can be encrypted at rest.
+To store credentials in a Vault instance, you can read the official Vault docs
+[Storing secrets in Vault](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-first-secret).
+
+You can also encrypt credentials at rest.
+you can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
```
Comment (Balowen) [inline docs/docs/secrets-managers.mdx:34]:
```suggestion
You can also encrypt credentials at rest.
You can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
```
Not sure if this should be two separate sentences, or one. If one then we should rephrase it somehow.

File: docs/docs/secrets-managers.mdx [hunk 33]
```diff
@@ -0,0 +1,152 @@
...
+and Rasa Pro will not be able to access the secrets.
+You will need to create a new renewable token and restart Rasa Pro with new token.
+:::
+
+
+### Authentication
+Rasa Pro can authenticate to Vault through
+[Token authentication](https://www.vaultproject.io/docs/auth/token).
+
+Both `expiring` and `non-expiring`(so called, root tokens) tokens are supported.
```
Comment (Balowen) [inline docs/docs/secrets-managers.mdx:54]:
```suggestion
Both `expiring` and `non-expiring` (so called, root tokens) tokens are supported.
```

File: docs/docs/secrets-managers.mdx [hunk 34]
```diff
@@ -0,0 +1,152 @@
...
+    # store secret in the Vault under the group rasa-secrets
+    vault kv put -mount=secret rasa-secrets sql_store_username=postgres sql_store_password=12345
+    ```
+
+3. Start Postgres container
+    ```bash
+    docker run --name postgres -e POSTGRES_DB=rasa -e POSTGRES_PASSWORD=12345 -p 5432:5432 -d postgres
+    ```
+
+4. [Configure tracker store with vault secerts manager](#how-to-configure-tracker-store-with-vault-secrets-manager)
```
Comment (Balowen) [inline docs/docs/secrets-managers.mdx:151]:
```suggestion
4. [Configure tracker store with vault secrets manager](#how-to-configure-tracker-store-with-vault-secrets-manager)
```

File: docs/docs/secrets-managers.mdx [hunk 35]
```diff
@@ -0,0 +1,142 @@
+---
+id: secrets-managers
+sidebar_label: Secrets Managers
```
[the remark this answers is not in the document]
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:3]: My 2 cents:
I originally named to feature Credentials Management but in the code I used Secrets Manager a lot. Mostly because I worked with AWS Secrets Manager in the past.
From code standpoint it would be easier to use Secrets Manager and to rename discovery ticket in JIRA.

@andreaklmn WDYT?

File: docs/docs/secrets-managers.mdx (hunk 32 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:34]: @Balowen 
I updated this to 
```
You can also encrypt credentials at rest with [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
```
WDYT?

File: docs/docs/secrets-managers.mdx (hunk 28 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx]: @Balowen 
I have updated the example to be more comprehensive.
Let me know what do you think.

File: docs/docs/secrets-managers.mdx (hunk 35 above)
[the remark this answers is not in the document]
Comment (m-vdb) [reply docs/docs/secrets-managers.mdx:3]: I think we should name it in a way that makes sense for our customers and our overall messaging. If both are equivalent, I agree with what you said and we should pick the simpler choice for us. But only if they're equivalent :D

File: docs/docs/secrets-managers.mdx (hunk 32 above)
Comment (Balowen) [reply docs/docs/secrets-managers.mdx]: LGTM

File: docs/docs/secrets-managers.mdx (hunk 35 above)
[the remark this answers is not in the document]
Comment (andreaklmn) [reply docs/docs/secrets-managers.mdx:3]: I guess for me the question is: are we managing all types of sensitive data, including passwords, API keys, encryption keys, certificates, and other types of confidential information - if yes, then it's "secrets management"

If we are only managing credentials, I would probably stick to credentials management..? Unless we are planning on managing other "secrets" as part of this solution

My 2 cents!

[the remark this answers is not in the document]
Comment (andreaklmn) [reply docs/docs/secrets-managers.mdx:3]: I just saw that the Vault website here uses the 2 terms somewhat interchangeably: https://www.vaultproject.io/
Under "automate credentials rotation" - so I'm down for using secrets management!

File: docs/docs/secrets-managers.mdx (hunk 25 above)
Comment (ancalita) [reply docs/docs/secrets-managers.mdx:29]: Perfect 👍🏻

File: docs/docs/secrets-managers.mdx (hunk 27 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx]: @Balowen Could you take a look at my comment above?

File: docs/docs/secrets-managers.mdx (hunk 35 above)
[the remark this answers is not in the document]
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx:3]: For anyone stumbling on this PR: We have discussed this and agreed to use Secrets Manager.
Just to put my answer here as well, we are able to manage sensitive data.

File: docs/docs/secrets-managers.mdx (hunk 27 above)
Comment (Balowen) [reply docs/docs/secrets-managers.mdx]: Looks good now

File: docs/docs/secrets-managers.mdx (hunk 28 above)
Comment (Balowen) [reply docs/docs/secrets-managers.mdx]: I think it's better now but I also agree with @l2nguyen [here](https://github.com/RasaHQ/rasa/pull/12110#discussion_r1132194604). It's unlikely our customers will ever run it on local.

**Comment by radovanZRasa** [comment]
@l2nguyen 
> Customers would definitely want an example on how to set up the configs for this in a kubernetes environment.

This depends on how does the CSE or infra team plan to make this available in the helm charts.
I do not see that we have done this for Event Brokers or Tracker Store. Is there a new approach that we should be familiar with?

**Comment by radovanZRasa** [comment]
@Balowen 
> That's a very good point. We do not have examples for Event Brokers or Tracker Store but they are fully supported by our helm charts.

I suggest that we get back to this (Vault included) in next couple of weeks as this cannot and should not be resolved on a PR level. We need to comprise a systematic approach on how we are gonna handle this in our docs and update docs for related features.

File: docs/docs/secrets-managers.mdx (hunk 22 above)
Comment (radovanZRasa) [reply docs/docs/secrets-managers.mdx]: I have remove the example section.

**Comment by radovanZRasa** [comment]
> Does that mean that for now it is not possible to use Secrets Manager with our current helm charts? If that's the case this would be another PR, but we definitely need to support Vault integration in our helm charts. Once we do support it we will be able to think about examples on how to configure it or whether we actually need to add such examples to our docs

We just need to add support for Secrets Manager to helm charts. I have spoken with @miraai about this. It was understood that we (Atom team) need to add support for Secrets Manager in our helm charts.
Nevertheless this should not be a blocker for Secrets Manager documentation.

@m-vdb I created a [ticket](https://rasahq.atlassian.net/browse/ATO-874) to add support for Secrets Manager on our helm chart.

**Review by m-vdb — APPROVED** [review]
🚀
